### PR TITLE
Replace hardcoded layer values

### PR DESCRIPTION
**Pull request by lubej**

### CODE REVIEW

**Review by buberdds — COMMENTED** [review]
I feel we can achieve correct links just by using existing router API.
Currently, to generate simple links we have useLayer, useLayerHref (useHref wrapper), RouteUtils, useParams. RouteUtils sometimes includes layer sometimes not. Having abstract class that has method that returns string `blocks` where it is used only in one place looks like overkill. API is not consistent 
`to={useLayerHref(RouteUtils.getDashboardRoute())}`
`to={RouteUtils.getTransactionRoute(transaction.hash, layer)}`
can't this be achieved by using one pattern ?

File: src/app/components/Account/AccountLink.tsx [hunk 1]
```diff
@@ -4,13 +4,14 @@ import { useTheme } from '@mui/material/styles'
...
-import { Layer } from '../../../config'
 import { RouteUtils } from '../../utils/route-utils'
+import { useLayer } from '../../hooks/useLaye'
 
-export const AccountLink: FC<{ address: string; paratime: Layer }> = ({ address, paratime }) => {
+export const AccountLink: FC<{ address: string }> = ({ address }) => {
   const theme = useTheme()
   const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
-  const to = RouteUtils.getAccountRoute(address, paratime)
+  const layer = useLayer()
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/components/Account/AccountLink.tsx:13]: The way the routes are structured now, we can't get loader data from `useLoaderData`, as it returns the nearest route loader data. And the layer param is one level above. Not sure if there is a way of bypassing this.

[the remark this answers is not in the document]
Comment (buberdds) [reply src/app/components/Account/AccountLink.tsx:13]: `useRouteLoaderData`

File: src/app/hooks/useLayerHref.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export const useLayerHref = (href = '', layer?: Layer): ReturnType<typeof useHref> => {
+  const layerParam = useLayer()
+
+  const getLayerPrefix = () => {
+    if (!href.startsWith('/')) {
+      return ''
+    }
+
+    if (layer) {
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/hooks/useLayerHref.ts:13]: Not yet, but I image adding Sapphire support on search page might need this.

File: src/app/components/Account/AccountLink.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/components/Account/AccountLink.tsx:13]: Got it, with adding the routeId param to the root route. Think it is cleaner this way.

**Comment by lubej** [comment]
> I feel we can achieve correct links just by using existing router API. Currently, to generate simple links we have useLayer, useLayerHref (useHref wrapper), RouteUtils, useParams. RouteUtils sometimes includes layer sometimes not. Having abstract class that has method that returns string `blocks` where it is used only in one place looks like overkill. API is not consistent `to={useLayerHref(RouteUtils.getDashboardRoute())}` `to={RouteUtils.getTransactionRoute(transaction.hash, layer)}` can't this be achieved by using one pattern ?

Right, so the issue with `useLayerHref`, is that it can't be used in scenarios like table rows. So if we want to unify it, we should go with `RouteUtils`. But with `RouteUtils` there is a downside, because you need to always pass in a layer param.

**Review by csillag — APPROVED** [review]
Reviewed. LGTM.

**Pull request merged**